litebird_sim/simulations.py

```diff
@@ -209,3 +233,66 @@ def flush(self):
             self.base_path / "report.html", "w", encoding="utf-8",
         ) as outf:
             outf.write(html_full_report)
+
+    def create_observations(
+        self,
+        detectors: List[Detector],
+        num_of_obs_per_detector: int,
+        start_time,
+        duration_s: float,
+        distribute=True,
+        use_mjd=False,
+    ):
+        "Create a set of Observation objects"
+
+        observations = []
+        for detidx, cur_det in enumerate(detectors):
+            cur_sampfreq_hz = cur_det.sampfreq_hz
+            num_of_samples = cur_sampfreq_hz * duration_s
+            samples_per_obs = distribute_evenly(num_of_samples, num_of_obs_per_detector)
+
+            if isinstance(start_time, float):
+                cur_time = start_time
+            else:
+                if use_mjd:
+                    cur_time = start_time
+                else:
+                    cur_time = start_time.mjd
+
+            for cur_obs_idx in range(num_of_obs_per_detector):
+                nsamples = samples_per_obs[cur_obs_idx].num_of_elements
+                cur_obs = Observation(
+                    detector=cur_det,
+                    start_time=cur_time,
+                    sampfreq_hz=cur_sampfreq_hz,
+                    nsamples=nsamples,
+                    use_mjd=use_mjd,
+                )
+                observations.append(cur_obs)
+                span_s = cur_sampfreq_hz * nsamples
+
+                if use_mjd:
+                    cur_time = Time(mjd=cur_time) + TimeDelta(seconds=span_s)
+                else:
+                    cur_time += span_s
+
+        if distribute:
+            self.distribute_workload(observations)
+
+        return observations
+
+    def distribute_workload(self, observations: List[Observation]):
+        if self.mpi_size == 1:
+            self.observations = observations
+            return
+
+        cur_rank = self.mpi_rank
+        span = distribute_optimally(
+            elements=observations,
+            num_of_groups=self.mpi_size,
+            weight_fn=lambda obs: obs.nsamples,
```

Comment:
Can you make an example where it is convenient to pass a weight functions instead of the weights directly?

Reply:
The example you highlighted (lines 290-293) is the perfect example: we pass a list of `Observation` objects, where the weight is not the object itself but rather its `nsamples` field. This line is the reason why I added the `weight_fn` parameter to the functions in this module.